bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java

```diff
@@ -0,0 +1,572 @@
+
+					XmlRpcStruct response = loadPowerDogResponse(serverNeedsUpdate, server);
+		
+					// update item state
+					if(response != null) 
+					{
+						String value = getVariable(response, provider.getValueId(itemName), provider.getName(itemName));
+						if (value != null) {
+							Class<? extends Item> itemType = provider.getItemType(itemName);
+							State state = createState(itemType, value);
+							eventPublisher.postUpdate(itemName, state);
+							lastUpdateMap.put(itemName, System.currentTimeMillis());
+						}
+					}
+				}
+			}
+		}
+
+		logger.debug("execute() method is finished!");
+	}
+
+
+	private XmlRpcStruct loadPowerDogResponse(boolean needsUpdate, PowerDogLocalApiServerConfig server) 
+	{
+		// Get all current linear values from the powerdog in case of an update
+		XmlRpcStruct response = null;
+		if(needsUpdate == true) 
+		{
+			  try {
+					logger.debug("PowerDogLocalApi querying PowerDog");
+			        
+					// perform XML RPC call and store response
+					PowerDog powerdog = ( PowerDog ) XmlRpcProxy.createProxy( server.url(), "", new Class[] { PowerDog.class }, false );
+				    response = powerdog.getAllCurrentLinearValues(server.password);
+				    server.cache = response;
+				    server.lastUpdate = System.currentTimeMillis();
+				    
+			        logger.debug("PowerDog.getAllCurrentLinearValues() result: " + response.toString());
+			     }
+				 catch (Exception e) 
+				 {
+					 logger.warn("PowerDogLocalApi querying PowerDog failed");
+					 logger.warn(e.getMessage());
+			     }	
+		  }
+		  else {
+			logger.debug("Using PowerDogLocalApi cache");
+			response = server.cache;
+		  }
+		return response;
+	}
+
+	/**
+	 * Parse PowerDog xmlrpc response to getAllCurrentLinearValues
+	 * @param response PowerDog Response
+	 * @param valueId Value ID of PowerDog Item
+	 * @param name Parameter name to be updated
+	 * @return
+	 */
+	private String getVariable(XmlRpcStruct response, String valueId, String name) 
+	{
+		try
+		{
+			XmlRpcStruct reply = response.getStruct("Reply");
+			XmlRpcStruct item  = reply.getStruct(valueId);
+			String value = item.getString(name);
+			return value;
+		}
+		catch (Exception e) 
+		{
+			return null;
+		}
+	}
+
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		logger.debug("internalReceiveCommand({},{}) is called!", itemName, command);
+		
+		State newState = null;
+
+		// cast Interfaces
+		if(command instanceof OnOffType) 			{ newState = (OnOffType) command; }
+		else if(command instanceof OpenClosedType) 	{ newState = (OpenClosedType) command; }
+		else if(command instanceof PercentType) 	{ newState = (PercentType) command; }
+		else if(command instanceof DecimalType) 	{ newState = (DecimalType) command; }
+		
+		if(newState != null)
+		{
+			eventPublisher.postUpdate(itemName, newState);
+		}
+		else
+		{
+			logger.warn("internalReceiveCommand({},{}) not handled!", itemName, command);
+		}
+	}
```

Comment:
i am not sure about the purpose of this method? Since internalReceiveCommand is called due to events being posted through the EventPublisher seem to produce a loop, right? I would suggest to remove this as this method is intended to communicate openHAB internal commands to the external hardware.